Added multiple server support and Server Name accepts all characters

MachinegunMarty opened the pull request:
…in the name now.

# Description

Removed the Querry Port from the startup command. (Ark Ascended doesn't use querry port anymore)
Changed Port={{SERVER_PORT}} to Port={{GAME_PORT}}
Added support for all characters in ServerName in the SessionName command, by using 3 backslashes instead of 1.
Added a Game Port (Default: 7777) in the panel to host more then 1 server
Changed the default RCON Port to 27020
Changed Max Players default from 70 to 50
Changed the description of BattlEye and set the default to 0 (NoBattlEye)

Edited the README.md to reflect the changes, how to setup a server, which ports to open and how to host multiple servers from the same ip.

## Checklist for all submissions

<!-- insert X into the brackets to mark it as done (i.e. [x]). You can click preview to make the links appear clickable. -->

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

<!-- If this is an egg update fill these out -->

* [x] You verify that the start command applied does not use a shell script
  * [ ] If some script is needed then it is part of a current yolk or a PR to add one
* [x] The egg was exported from the panel

<!-- You can erase the new egg submission template if you're not adding a completely new egg -->

parkervcp commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -111,10 +111,10 @@
-            "description": "ARK query port used by steam server browser and ark client server browser.",
-            "env_variable": "QUERY_PORT",
-            "default_value": "27015",
+            "name": "Game Port",
```
This is already set by the primary allocation This is not needed.

MachinegunMarty replied:
The SERVER_NAME does the same indeed. I just thought to make it more clear.

parkervcp commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -154,7 +154,7 @@
             "description": "required for console commands and proper server shutdown (saveworld\/exit), doesn't have to be open to the public",
             "env_variable": "RCON_PORT",
-            "default_value": "37015",
+            "default_value": "27020",
```
There is also no reason to change the default port. It's internal only unless you want it exposed outside the container.

MachinegunMarty replied:
Doesn't have to be changed, can be any port anyways.
Ark Survival Evolved used 27020 for RCON, again just to make it clearer for people coming from Ark Survival Ascended.

parkervcp commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -15,7 +15,7 @@
     },
     "file_denylist": [],
-    "startup": "rmv() { echo \"stopping server\"; rcon -t rcon -a 127.0.0.1:${RCON_PORT} -p ${ARK_ADMIN_PASSWORD} KeepAlive && rcon -t rcon -a 127.0.0.1:${RCON_PORT} -p ${ARK_ADMIN_PASSWORD} DoExit && wait ${ARK_PID}; echo \"Server Closed\"; exit; }; trap rmv 15 2; proton run .\/ShooterGame\/Binaries\/Win64\/ArkAscendedServer.exe {{SERVER_MAP}}?listen?MaxPlayers={{MAX_PLAYERS}}?SessionName=\"{{SE ...
+    "startup": "rmv() { echo \"stopping server\"; rcon -t rcon -a 127.0.0.1:${RCON_PORT} -p ${ARK_ADMIN_PASSWORD} KeepAlive && rcon -t rcon -a 127.0.0.1:${RCON_PORT} -p ${ARK_ADMIN_PASSWORD} DoExit && wait ${ARK_PID}; echo \"Server Closed\"; exit; }; trap rmv 15 2; proton run .\/ShooterGame\/Binaries\/Win64\/ArkAscendedServer.exe {{SERVER_MAP}}?listen?MaxPlayers={{MAX_PLAYERS}}?SessionName=\\\"{{ ...
```
As called out in the lower section. The `GAME_PORT` isn't needed as the `SERVER_PORT` is the default allocation.

MachinegunMarty replied:
As explained, just to make it more clear to people.

Do i need to make adjustments and make a new pr or how does that work ?

QuintenQVD0 replied:
No you append them to this one, The pirmary port gets automaticly set in the variable SERVER_PORT and should never be filled in any other variable

MachinegunMarty replied:
I applied the changes to my branch for review.

QuintenQVD0 commented:
You pushed way to mutch @MachinegunMarty

QuintenQVD0 commented:
And again you did not fully export the egg what is not allowed, make changes in the panel, export and fully replace the egg here!

MachinegunMarty commented:
> And again you did not fully export the egg what is not allowed, make changes in the panel, export and fully replace the egg here!

I forked the github, used Visual Studio to make a branch and adjusted stuff, and pushed the commit to my branch.

Is this not the way to do this ?

QuintenQVD0 commented:
> > And again you did not fully export the egg what is not allowed, make changes in the panel, export and fully replace the egg here!
> 
> I forked the github, used Visual Studio to make a branch and adjusted stuff, and pushed the commit to my branch.
> 
> Is this not the way to do this ?

Your vscode is configured wrong as by default it should be .vscode not .vs

Please upodate your local branch as I pushed changes to it to remove all those unneeded files, also Please update the egg fully again so the timestamp updates

QuintenQVD0 commented on game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json:
```diff
@@ -62,9 +62,9 @@
         },
         {
             "name": "Battle Eye",
-            "description": "Enable BattleEye \/ Anti-Cheat",
```
Please leave the comment and default value as they do not even see the 0 or 1 they see a toggle switch

MachinegunMarty replied:
This is my first pr so help me out please.

I downloaded the changes you made to the branch, but how to continue ?

Do i make adjustments to the files in my branch in VS ? 
Or do i need to upload the json in the panel and export it ?
I'm confused ?

QuintenQVD0 replied:
you can not touch the .json file with an editor, you must make the changes in a panel and then hit export!

MachinegunMarty commented:
I adjusted the original one in the portal and made an export.

- Only removed querry port from startup and as option
- Added backslash to the Session Name startup command.

[egg-ark--survival-ascended-export.json](https://github.com/parkervcp/eggs/files/13295699/egg-ark--survival-ascended-export.json)

MachinegunMarty commented:
My latest should be good now, i hope.
I downloaded the latest egg, imported it into my panel, made the adjustments, saved and exported the egg.
Then i copy / paste the exported egg into my branch, overwriting the original json. Commit and push.

dagbs commented:
I don't understand what you changed. You didn't address the multi-server issue... and it turns out it's not an "egg" issue, it's an environmental issue; so, the changes you've made are irrelevant to resolving any issues.

dagbs commented:
https://ark.wiki.gg/wiki/Dedicated_server_setup#Network

also, to note, the only thing I accidently messed up on was in the README of calling "Raw UDP" instead of "Peer" but since it's not something that's set in the panel, and is always +1 from the Game/Primary Port. While it's probably not used right now, it's in their Wiki and its most likely how Steam users will be able to "join off of each other" in the future once that is fixed.

dagbs commented:
The only part I would be interested in testing and may approve, if it works as expected, is the server name fix.
Everything else I'm against approving.

MachinegunMarty commented:
> The only part I would be interested in testing and may approve, if it works as expected, is the server name fix. Everything else I'm against approving.

I'll type out here what i did.

1. I changed `SessionName=\"{{SESSION_NAME}}\"?` into `SessionName=\\\"{{SESSION_NAME}}\\\"?` in startup
2. Removed `QueryPort={{QUERY_PORT}}?' from startup
3. Removed QuerryPort from the panel options

The querry port is not used anymore.

I'm hosting 2 servers from the same ip. Using port 7777 for server 1 and 7778 for server 2. 7777-7778 UDP are the only open ports in my router. Can log in to both servers.

What you do with it is up to you.
I only tried to help out

parkervcp commented:
> I only tried to help out, but will keep my shit to myself in the future.

I'm sorry you feel this way as I want my repo to be open to changes from everyone.

I still believe the only change really required here is the one pertaining to the server name. I still find it weird that it's needed but it could be a special character issue.
